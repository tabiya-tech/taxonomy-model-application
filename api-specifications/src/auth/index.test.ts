describe("Test the auth module", () => {
  test("The auth module can be required via the index", async () => {
    // GIVEN the module
    // WHEN the module is required via the index
    expect(async () => await import("./")).not.toThrow(); // We check that it doesn't throw an error instead of simply letting it fail on import because we want an easier error message
    const authModule = await import("./");

    // THEN check if Schema is defined in it
    expect(authModule.default.Schemas.Request.Context).toBeDefined();
  });

  test("The auth module matches the snapshot", () => {
    // GIVEN the module
    // WHEN the module is required via the index
    expect(async () => {
      // THEN Check if the module can be required without error
      const authModule = await import("./");
      expect(authModule.default).toMatchSnapshot();
    }).not.toThrowError();
  });
});

export {};
