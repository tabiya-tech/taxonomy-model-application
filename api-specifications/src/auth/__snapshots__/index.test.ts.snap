// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the auth module The auth module matches the snapshot 1`] = `
{
  "Enums": {
    "TabiyaRoles": {
      "ANONYMOUS": "anonymous",
      "MODEL_MANAGER": "model-managers",
      "REGISTERED_USER": "registered-user",
    },
  },
  "Schemas": {
    "Request": {
      "Context": {
        "$id": "/components/schemas/AuthResponseContextSchema",
        "additionalProperties": false,
        "properties": {
          "roles": {
            "description": "A comma-separated list of roles.",
            "maxLength": 4096,
            "type": "string",
          },
          "username": {
            "description": "The username of the user.",
            "maxLength": 256,
            "pattern": "\\S",
            "type": "string",
          },
        },
        "required": [
          "username",
        ],
        "type": "object",
      },
    },
  },
  "Types": {},
}
`;
